python/ray/data/read_api.py

```diff
@@ -698,6 +698,124 @@ def read_parquet(
+            implementation to read from. These filesystems are specified in the
+            `pyarrow docs <https://arrow.apache.org/docs/python/api/\
+            filesystems.html#filesystem-implementations>`_. Specify this parameter if
+            you need to provide specific configurations to the filesystem. By default,
+            the filesystem is automatically selected based on the scheme of the paths.
+            For example, if the path begins with ``s3://``, the ``S3FileSystem`` is
+            used. If ``None``, this function uses a system-chosen implementation.
+            If on databricks, this argument will be ignored.
+        columns: A list of column names to read. Only the specified columns are
+            read during the file scan.
+        parallelism: The amount of parallelism to use for the dataset. Defaults to -1,
+            which automatically determines the optimal parallelism for your
+            configuration. You should not need to manually set this value in most cases.
+            For details on how the parallelism is automatically determined and guidance
+            on how to tune it, see :ref:`Tuning read parallelism
+            <read_parallelism>`. Parallelism is upper bounded by the total number of
+            records in all the parquet files.
+        ray_remote_args: kwargs passed to :meth:`~ray.remote` in the read tasks.
+        tensor_column_schema: A dict of column name to PyArrow dtype and shape
+            mappings for converting a Parquet column containing serialized
+            tensors (ndarrays) as their elements to PyArrow tensors. This function
+            assumes that the tensors are serialized in the raw
+            NumPy array format in C-contiguous order (e.g., via
+            `arr.tobytes()`).
+        meta_provider: A :ref:`file metadata provider <metadata_provider>`. Custom
+            metadata providers may be able to resolve file metadata more quickly and/or
+            accurately. In most cases you do not need to set this parameter.
+        arrow_parquet_args: Other parquet read options to pass to PyArrow. For the full
+            set of arguments, see the`PyArrow API <https://arrow.apache.org/docs/\
+                python/generated/pyarrow.dataset.Scanner.html\
+                    #pyarrow.dataset.Scanner.from_fragment>`_
+
+    Returns:
+        :class:`~ray.data.Dataset` producing records read from the specified parquet
+        files.
+    """
+    from ray.util.spark.utils import (
+        convert_dbfs_path_to_local_path,
+        get_or_create_spark_session_with_delta_extension,
+        get_spark_session,
+        is_in_databricks_runtime,
+    )
+
+    if is_in_databricks_runtime():
+        spark = get_spark_session()
+        paths = spark.read.load(uri, format="delta").inputFiles()
+        paths = [convert_dbfs_path_to_local_path(path) for path in paths]
+    else:
+        spark = get_or_create_spark_session_with_delta_extension(delta_package)
```

Comment:
Will this run a Spark cluster in where the code is called? Can we let the user pass in the SparkSession object instead?

Reply:
oh sorry, this PR implementation has some issue, we are going to close it. we have a replacement PR https://github.com/ray-project/ray/pull/39852